ironfish/src/workerPool/pool.ts

```diff
@@ -117,9 +118,11 @@ export class WorkerPool {
     const queue = this.queue
 
     this.workers = []
-    this.queue = []
 
     queue.forEach((j) => j.abort())
+
+    queue.clearAll()
```

Comment:
I think if you wanted to make an `abortAll` function and replace the above `forEach` with that, it could be handy. The queue could also drop all of the jobs when it calls abort on them.

The clearAll is okay in this situation, but I'm concerned that if someone happens to call it without cleaning up the promises (calling abort), anything awaiting on the job's promise could get stuck forever. Or in other words, I don't think we'd ever want to clear the queue without also aborting the jobs, so it'd make sense to combine them.

Reply:
Good call. Initially was just trying to map 1:1 with our current usage of the queue, but that does make for a more brittle API. 👍